### PR TITLE
use client factory to create namespace

### PR DESCRIPTION
**Pull request by tatlat**
*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/clients/kubernetes/unauth.go [hunk 1]
```diff
@@ -41,6 +42,10 @@ func (c *UnAuthClient) Get(ctx context.Context, name, namespace, kubeconfig stri
 		return fmt.Errorf("getting kubernetes resource: %v", err)
 	}
 
+	if namespace == "" {
+		return c.kubectl.Get(ctx, resourceType, kubeconfig, obj, &KubectlGetOptions{Name: name, ClusterScoped: pointer.Bool(true)})
```
Comment (g-gaston) [inline pkg/clients/kubernetes/unauth.go:46]: Nit: can we just set the options conditionally and have only one get call

File: pkg/clients/kubernetes/unauth_test.go [hunk 2]
```diff
@@ -62,6 +63,14 @@ func TestUnAuthClientGetSuccess(t *testing.T) {
 				Name:      tt.name,
 				Namespace: tt.namespace,
 			}
+
+			if tt.namespace == "" {
+				o = &kubernetes.KubectlGetOptions{
```
Comment (g-gaston) [inline pkg/clients/kubernetes/unauth_test.go:68]: The options should probably be part of the table

File: pkg/workflows/create_prep.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/aws/eks-anywhere/pkg/workflows/interfaces"
+)
+
+// CreateNamespaceIfNotPresent creates the namespace on the cluster if it does not already exist.
+func CreateNamespaceIfNotPresent(ctx context.Context, namespace, kubeconfig string, clientFactory interfaces.ClientFactory) error {
```
Comment (g-gaston) [inline pkg/workflows/create_prep.go:14]: Nit: can we find a different place for this?
Im ok if you move it in a separate pr

**Comment by mitalipaygude** [comment]
looks good to me!

File: pkg/workflows/create_prep.go (hunk 3 above)
Comment (tatlat) [reply pkg/workflows/create_prep.go:14]: i'll do it in another pr
